app/src/main/java/com/abhijitvalluri/android/fitnotifications/services/NLService.java

```diff
@@ -296,28 +252,35 @@ public void onNotificationPosted(final StatusBarNotification sbn) {
               .append("] ");
         }
 
+        if (mTransliterateNotif) {
+            title = transliterate(title);
+            notificationText = transliterate(notificationText);
+            notificationBigText = transliterate(notificationBigText);
+        }
+
         sb.append(notificationText);
 
-        if (notificationBigText.length() > 0) {
+        if (!isBlank(notificationBigText)) {
             sb.append(" -- ").append(notificationBigText);
         }
 
         RemoteViews contentView = new RemoteViews(getPackageName(), R.layout.custom_notification);
         contentView.setTextViewText(
                 R.id.customNotificationText, getString(R.string.notification_text));
 
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(this);
         builder.setSmallIcon(R.drawable.ic_sms_white_24dp)
-                .setContent(contentView);
+                .setContent(contentView)
+                .setContentTitle(title)
+                .setLocalOnly(true);        // avoid bridging this notification to other devices
 
-        if (mTransliterateNotif) {
-            try {
-                builder.setContentTitle(Transliterator.getInstance("Any-Latin")
-                        .transform(extras.getCharSequence(Notification.EXTRA_TITLE).toString()));
-            } catch (NullPointerException e) {
-                builder.setContentTitle(extras.getCharSequence(Notification.EXTRA_TITLE));
-            }
-        } else {
-            builder.setContentTitle(extras.getCharSequence(Notification.EXTRA_TITLE));
+        // use minimal priority for placeholder notifications
+        builder.setPriority(NotificationCompat.PRIORITY_MIN);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            // prevent notification from appearing on the lock screen
+            // (also avoid revealing sensitive data)
+            builder.setVisibility(Notification.VISIBILITY_SECRET);
```

Comment:
Why do you think someone might prefer to have a placeholder notification (without any actual information) to appear on the lock screen?
After all we are duplicating some other notification. And that original one is already there, on the lock screen.